Objects/longobject.c

```diff
@@ -3740,7 +3690,7 @@ k_mul(PyLongObject *a, PyLongObject *b)
     /* Split a & b into hi & lo pieces. */
     shift = bsize >> 1;
     if (kmul_split(a, shift, &ah, &al) < 0) goto fail;
-    assert(Py_SIZE(ah) > 0);            /* the split isn't degenerate */
+    assert(_PyLong_UnsignedDigitCount(ah) > 0);            /* the split isn't degenerate */
```

Comment:
This should just check the sign, right?
```suggestion
    assert(_PyLong_IsPositive(ah));            /* the split isn't degenerate */
```
Same below several occurrences.

Reply:
Done